google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,251 @@
+
+    Returns:
+    The pattern string, escaped except for the "*", which is
+    transformed into ".+" (match on one or more characters).
+    """
+    return '^{}$'.format(re.escape(pattern_string).replace('\\*', '.+'))
+
+
+class InstanceNetworkInterfaceRulesEngine(bre.BaseRulesEngine):
+    """Rules engine for InstanceNetworkInterfaceRules."""
+
+    def __init__(self, rules_file_path, snapshot_timestamp=None):
+        """Initialize.
+
+        Args:
+            rules_file_path: file location of rules
+        """
+        super(InstanceNetworkInterfaceRulesEngine,
+              self).__init__(rules_file_path=rules_file_path)
+        self.rule_book = None
+
+    def build_rule_book(self, global_configs=None):
+        """Build InstaceNetworkInterfaceRuleBook from rules definition file."""
+        self.rule_book = InstaceNetworkInterfaceRuleBook(
+            self._load_rule_definitions())
+
+    # pylint: disable=arguments-differ
+    def find_policy_violations(self, instance_network_interface,
+                               force_rebuild=False):
+        """Determine whether the networks violates rules."""
+        violations = itertools.chain()
+        if self.rule_book is None or force_rebuild:
+            self.build_rule_book()
+        resource_rules = self.rule_book.get_resource_rules()
+
+        for rule in resource_rules:
+            violations = itertools.chain(violations,
+                                         rule.
+                                         find_violations(
+                                             instance_network_interface))
+        return violations
+
+    def add_rules(self, rules):
+        """Add rules to the rule book."""
+        if self.rule_book is not None:
+            self.rule_book.add_rules(rules)
+
+
+class InstaceNetworkInterfaceRuleBook(bre.BaseRuleBook):
+    """The RuleBook for enforced networks resources."""
+
+    def __init__(self,
+                 rule_defs=None):
+        """Initialize.
+
+        Args:
+            rule_defs: The parsed dictionary of rules from the YAML
+                definition file.
+            snapshot_timestamp: The snapshot to lookup data.
+        """
+        super(InstaceNetworkInterfaceRuleBook, self).__init__()
+        self.resource_rules_map = {}
+        if not rule_defs:
+            self.rule_defs = {}
+        else:
+            self.rule_defs = rule_defs
+            self.add_rules(rule_defs)
+
+    def add_rules(self, rule_defs):
+        """Add rules to the rule book."""
+        print(rule_defs)
+        for (i, rule) in enumerate(rule_defs.get('rules', [])):
+            self.add_rule(rule, i)
+
+    def add_rule(self, rule_def, rule_index):
+        """Add a rule to the rule book.
+
+        Add a rule to the rule book.
+
+        The rule supplied to this method is the dictionary parsed from
+        the rules definition file.
+
+        For example, this rule...
+
+            # rules yaml:
+            rules:
```

Comment:
yeah, i meant the yaml, and i changed the rule format and then didnt update the doc